### DISCUSSION TITLE
Mullvad Qubes Proxy VM guide

### DISCUSSION THREAD

**Discussion by grrrrr** [PrivSec-dev/privsec.dev#63] [Q&A] [answered]
I'm working through the guide you recently [put out](https://privsec.dev/apps/using-mullvad-vpn-on-qubes-os/) and am running into DNS issues and I wonder if this is something that needs specific configuration to get it working.

- I have AppVM going through the Mullvad ProxyVPN.
- With VPN off AppVM can reach the internet (not over VPN)
- When open Mullvad GUI client the VPN on ProxyVPN sets up a wireguard connection 
- On the AppVM I can briefly connect to websites once wireguard is connected, after that DNS starts failing (I can reach sites by IP)

How does the AppVM get a working DNS setup,  looking at the Qubes Manager it shows it is using the Qubes system wide DNS resolvers instead of a Mullvad DNS server/

**Comment by TommyTran732** [comment, marked as answer]
Did you add the workaround 

```
echo "sleep 10 # Waiting a bit that Mullvad succeeds to establish connection
/usr/lib/qubes/qubes-setup-dnat-to-ns" | sudo tee -a /rw/config/rc.local
```

in the ProxyVM? And did you enable the Mullvad daemon and set Mullvad to autoconnect at boot?

**Reply by grrrrr** [reply]
I am not understanding what I did incorrect the first time but it is working,  I suspect I might have done some of the changes in a running AppVM terminal instead of the Template

**Reply by TommyTran732** [reply]
Awesome. Sounds like you sorted it out.
